src/middleware/devtools.ts

```diff
@@ -147,13 +148,41 @@ type DevtoolsImpl = <T>(
 
 export type NamedSet<T> = WithDevtools<StoreApi<T>>['setState']
 
+type Connection = ReturnType<
+  NonNullable<Window['__REDUX_DEVTOOLS_EXTENSION__']>['connect']
+>
+type ConnectionName = string | undefined
+type StoreName = string
+type StoreInformation = {
+  api: StoreApi<unknown>
+  initialState: unknown
+}
+type ConnectionInformation = {
+  connection: Connection
+  stores: Record<StoreName, StoreInformation>
+}
+const trackedConnections: Map<ConnectionName, ConnectionInformation> = new Map()
+
+const getTrackedConnectionState = (
+  name: string | undefined
+): Record<string, any> => {
+  const api = trackedConnections.get(name)
+  if (!api) return {}
+  return Object.fromEntries(
+    Object.entries(api.stores).map(([key, { api }]) => [key, api.getState()])
+  )
+}
+
 const devtoolsImpl: DevtoolsImpl =
   (fn, devtoolsOptions = {}) =>
   (set, get, api) => {
-    type S = ReturnType<typeof fn>
+    const { enabled, anonymousActionType, store, ...options } = devtoolsOptions
+
+    type S = ReturnType<typeof fn> & {
+      [store: string]: S
+    }
```

Comment:
Sure about this type? Seems very weird.

Reply:
I agree, though it's tricky to get rid of entirely. I _did_ manage to get it working without recursively referring to S, at least. @pavlobu might be able to explain it more; I was basically just going with it when doing the initial changes.